### PR TITLE
Vca rewards

### PR DESCRIPTION
**Pull request by danielSanchezQ**
It could have been calculated directly in the excel file with a simple formula too.

### CODE REVIEW

File: src/rewards/veterans.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    name: VeteranAdvisorId,
+    #[serde(alias = "No. of Reviews")]
+    number_of_reviews: usize,
+}
+
+pub fn calculate_veteran_advisors_rewards(
+    veteran_reviews: &[VeteranReview],
+    base_rewards: VeteranAdvisorReward,
+) -> VeteranAdvisorRewards {
+    let total_reviews: VeteranAdvisorReward = VeteranAdvisorReward::from(
```
Comment (Zeegomo) [inline src/rewards/veterans.rs:19]: nit: type and variable names seems to disagree on what the data represents (also the type specification is redundant). I would leave this as `usize` and do the conversion later for better readability

File: src/rewards/veterans.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+use serde::Deserialize;
+
+pub type VeteranAdvisorId = String;
+pub type VeteranAdvisorReward = rust_decimal::Decimal;
+
+pub type VeteranAdvisorRewards = Vec<(VeteranAdvisorId, VeteranAdvisorReward)>;
+
+#[derive(Deserialize)]
+pub struct VeteranReview {
```
Comment (Zeegomo) [inline src/rewards/veterans.rs:9]: nit: naming is a bit confusing. This is not actually a review, but rather holds info about a vCA

File: src/utils/csv.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+use serde::Serialize;
+
+use std::path::Path;
+
+pub fn load_data_from_csv<T: DeserializeOwned>(file_path: &Path) -> Result<Vec<T>, csv::Error> {
+    let mut csv_reader = csv::ReaderBuilder::new()
+        .has_headers(true)
+        .from_path(file_path)?;
+    let mut res = Vec::new();
+    for entry in csv_reader.deserialize() {
```
Comment (Zeegomo) [inline src/utils/csv.rs:11]: nit: I think it should be possible to do something like `csv_reader.deserialize().collect::<Result<Vec<T>, _>>()?`

File: src/rewards/veterans.rs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            .map(|vr| vr.number_of_reviews)
+            .sum::<usize>(),
+    );
+
+    veteran_reviews
+        .iter()
+        .map(|vr| {
+            (
+                vr.name.clone(),
+                (VeteranAdvisorReward::from(vr.number_of_reviews) / total_reviews) * base_rewards,
```
Comment (Zeegomo) [inline src/rewards/veterans.rs:31]: It just occurred to me that this (and similar other calculations in catalyst-toolbox) returns a decimal value, whereas in the end we need to send an integer amount of lovelaces. Is the conversion done manually?

File: src/rewards/veterans.rs (hunk 2 above)
Comment (danielSanchezQ) [reply src/rewards/veterans.rs:9]: Indeed, let me find a more suitable naming for this.

File: src/utils/csv.rs (hunk 3 above)
Comment (danielSanchezQ) [reply src/utils/csv.rs:11]: I would love for it to, but sadly it is not iirc.

**Review by Zeegomo — APPROVED** [review]
We probably should add a simple test for this (and open a PR from the upstream repo branch so that CI can run)

**Comment by danielSanchezQ** [comment]
@Zeegomo agree, and damn upstream branches 😆

**Comment by danielSanchezQ** [comment]
migrated to #69

**Pull request closed without merging**